### ISSUE TITLE
Precision at k computation confusion

### ISSUE THREAD

**Issue by tomasvanoyen** [lyst/lightfm#500]
Hi, 

I have some confusion on the computation of the precision at k metric (lightfm version 1.15). 

In particular, I understand from the code in evaluation.py, the precision at k metric describes the fraction of known positives in the first k positions of the ranked list of results. What is unclear to me is how the training and test data points are untangled when computing the rank. 

Below, I demonstrate my confusion considering the Movielens example. 

Using the code provided at  the example [docs](https://lyst.github.io/lightfm/docs/examples/movielens_implicit.html), we can validate the model as follows: 

`from lightfm.datasets import fetch_movielens`
`movielens = fetch_movielens()`
`train = movielens['train']`
`test = movielens['test']`
`model_rcv = LightFM(loss='bpr', random_state=2016, learning_rate=0.05)`
`model_rcv.fit(train, epochs=10)`

The precision at k, can now be computed as: 

`print("Train precision: {}".format(precision_at_k(model_rcv, train,k=10).mean()))`
`print("Test precision: {}".format(precision_at_k(model_rcv, test,k=10).mean()))`
Train precision: 0.585684
Test precision: 0.09586427

When I now try to understand this result, and look at the predicted test rankings by considering: 
`ranks = model_rcv.predict_rank(test,train_interactions=train)`

I am surprised to see that `ranks.tocoo().data.max()` returns 
1642.0

Am I correct that this means that the lowest predicted rank among the test interactions is 1642? And am I correct that this implies that the item/user interactions are ranked also with respect to item/user interactions which are in the training set? 

Looking forward to your reply, 

Best regards, 

Tomas

**Comment by SimonCW** [comment]
I’m closing this issue because it has been inactive for a long time. If you still encounter the problem, please open a new issue.

Thank you!
